src/main/java/org/opensearch/knn/index/codec/KNN91Codec/KNN91Codec.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.knn.index.codec.KNN91Codec;
+
+import org.apache.lucene.codecs.Codec;
+import org.apache.lucene.codecs.CompoundFormat;
+import org.apache.lucene.codecs.DocValuesFormat;
+import org.apache.lucene.codecs.FilterCodec;
+import org.apache.lucene.codecs.lucene91.Lucene91Codec;
+import org.opensearch.knn.index.codec.KNNDocFormatFacade;
+import org.opensearch.knn.index.codec.KNNDocFormatFactory;
+
+/**
+ * Extends the Codec to support a new file format for KNN index
+ * based on the mappings.
+ *
+ */
+public final class KNN91Codec extends FilterCodec {
+
+    public static final String KNN_91 = "KNN91Codec";
+    private KNNDocFormatFacade docFormatFacade;
```

Comment:
Hmmm I think a different name would be good other than KNNDocFormatFacade - maybe KNNFormatFacade?

Reply:
Agree, KNNFormatFacade is more reasonable